sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1064 @@
+ * interface the necessity for null checks can easily be deducted from the annotations.
+ * </p>
+ * <p>
+ * The Jsr305AnnotationsCheck supports enforcing the following code style:
+ * </p>
+ * <ul>
+ * <li>Every method declaration, implementation or lambda requires nullness annotations for all
+ * parameters and return values except for primitive types (because a void or an int can never be
+ * null anyway).</li>
+ * <li>The annotation can be made directly or applied through either inheritance from an already
+ * annotated class or a annotation for class-wide defaults.</li>
+ * <li>Annotations need to make sense. For instance, a class-scope annotation cannot be used for a
+ * method, and a method annotation cannot be used for a class.</li>
+ * <li>In overridden methods, the following rule applies (regardless of what was annotated in the
+ * parent method): For parameter definitions {@code @Nonnull} annotation is always illegal because
+ * being less "nullable" cannot be assumed for a parameter in an inherited method. Conversely
+ * {@code @Nullable} is always allowed. For return values it is the other way round:
+ * {@code @CheckForNull} is always illegal while {@code @Nonnull} is always legal.</li>
+ * </ul>
+ * <p>
+ * The following configuration properties are supported:
+ * </p>
+ * <dl>
+ * <dt>{@code packages = com.github.sevntu.pkg1,com.github.sevntu.pkg2}</dt>
+ * <dd>Activate this check for a list of parent packages and their children.</dd>
+ * <dt>{@code excludePackages = com.github.sevntu.pkg1.sub1,com.github.sevntu.pkg1.sub2}</dt>
+ * <dd>Set packages excluded from checking. This setting can be useful if under the parent package
+ * set with "packages" there are subpackages which should not be checked.</dd>
+ * <dt>{@code allowOverridingReturnValue = true}</dt>
+ * <dd>Annotating return values "@CheckForNull" in overridden methods is flagged as an error. When
+ * setting the this property to true, this error will be ignored (useful for upgrading).</dd>
+ * <dt>{@code allowOverridingParameters = true}</dt>
+ * <dd>Annotating parameters "@Nonnull" in overridden methods is flagged as an error. When setting
+ * this property to true, this error will be ignored (useful for upgrading).</dd>
+ * </dl>
+ * <p>
+ * Example code:
+ * </p>
+ *
+ * <pre>
+ * // no class-level annotation
+ * class Class1 {
+ *     &#64;CheckForNull // Error: obj2 not annotated!
+ *     String method1(@Nullable Object obj1, Object obj2) { ... }
+ *
+ *     // Error: return value not annotated
+ *     String method2() { ... }
+ * }
+ *
+ * // with class-level defaults
+ * &#64;ParametersAreNonnullByDefault
+ * class Class2 {
+ *     &#64;CheckForNull // Legal
+ *     String method1(Object obj1, Object obj2) { ... }
+ *
+ *     &#64;Nonnull // Legal
+ *     String method2(Object obj1, @Nullable Object obj2) { ... }
+ *
+ *     &#64;Nonnull // Error, redefinition of obj2's nullness
+ *     String method3(Object obj1, @Nonnull Object obj2) { ... }
```

Comment:
please remove "..." it better to keep compilable.

Reply:
Done with next commit.